packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -301,43 +301,38 @@ class CupertinoDatePicker extends StatefulWidget {
     this.itemExtent = _kItemExtent,
     this.selectionOverlayBuilder,
   }) : initialDateTime = initialDateTime ?? DateTime.now(),
-       assert(
-         itemExtent > 0,
-         'item extent should be greater than 0',
-       ),
-       assert(
-         minuteInterval > 0 && 60 % minuteInterval == 0,
-         'minute interval is not a positive integer factor of 60',
-       ) {
-    assert(
-      mode != CupertinoDatePickerMode.dateAndTime || minimumDate == null || !this.initialDateTime.isBefore(minimumDate!),
-      'initial date is before minimum date',
-    );
-    assert(
-      mode != CupertinoDatePickerMode.dateAndTime || maximumDate == null || !this.initialDateTime.isAfter(maximumDate!),
-      'initial date is after maximum date',
-    );
-    assert(
-      (mode != CupertinoDatePickerMode.date && mode != CupertinoDatePickerMode.monthYear) || (minimumYear >= 1 && this.initialDateTime.year >= minimumYear),
-      'initial year is not greater than minimum year, or minimum year is not positive',
-    );
-    assert(
-      (mode != CupertinoDatePickerMode.date && mode != CupertinoDatePickerMode.monthYear) || maximumYear == null || this.initialDateTime.year <= maximumYear!,
-      'initial year is not smaller than maximum year',
-    );
-    assert(
-      (mode != CupertinoDatePickerMode.date && mode != CupertinoDatePickerMode.monthYear) || minimumDate == null || !minimumDate!.isAfter(this.initialDateTime),
-      'initial date ${this.initialDateTime} is not greater than or equal to minimumDate $minimumDate',
-    );
-    assert(
-      (mode != CupertinoDatePickerMode.date && mode != CupertinoDatePickerMode.monthYear) || maximumDate == null || !maximumDate!.isBefore(this.initialDateTime),
-      'initial date ${this.initialDateTime} is not less than or equal to maximumDate $maximumDate',
-    );
-    assert(
-      this.initialDateTime.minute % minuteInterval == 0,
-      'initial minute is not divisible by minute interval',
-    );
-  }
+      assert(itemExtent > 0, 'item extent should be greater than 0'),
```

Comment:
Not to come in late and suggest a change, but I wonder if it would be better to change the argument type in the constructor of `showDayOfWeek` to `bool? showDayOfWeek`, then add this above the asserts:
```suggestion
      showDayOfWeek = mode == CupertinoDatePickerMode.date ? showDayOfWeek ?? false : showDayOfWeek,
      assert(itemExtent > 0, 'item extent should be greater than 0'),
```
Then in the assert itself, check if for date picker types that don't need it if the property is not null. That way we can maintain the behavior of using a default of false, while still alerting the developer that they are trying to set a property that won't do anything, for both true and false.